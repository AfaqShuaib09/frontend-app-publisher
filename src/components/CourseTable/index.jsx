import React from 'react';

import TableContainer from '../../containers/TableContainer';
import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';

const CourseTable = () => {
  const courseTableColumns = [
    {
      label: 'Course Name',
      key: 'title',
      columnSortable: true,
    },
    {
      label: 'Course Number',
      key: 'key',
      columnSortable: true,
    },
    {
      label: 'Owner',
      key: 'owners',
      columnSortable: false,
    },
    {
      label: 'Modified',
      key: 'modified',
      columnSortable: true,
    },
  ];

  const formatCourseData = courses => courses.map(course => ({
    ...course,
    owners: course.owners ? course.owners.map(owners => owners.name).join(', ') : '',
  }));

  return (
    <TableContainer
      id="courses"
      className="courses"
      fetchMethod={DiscoveryDataApiService.fetchCourses}
      columns={courseTableColumns}
      formatData={formatCourseData}
      tableSortable
    />
  );
};

export default CourseTable;
