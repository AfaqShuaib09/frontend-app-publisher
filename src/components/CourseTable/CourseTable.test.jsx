import React from 'react';
import { shallow } from 'enzyme';

import CourseTable from './index';

describe('CourseTable', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<CourseTable debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render with the expected "Hello Publisher" text', () => {
    const component = shallow(<CourseTable />);
    expect(component.text()).toEqual('Hello Publisher!');
  });
});
