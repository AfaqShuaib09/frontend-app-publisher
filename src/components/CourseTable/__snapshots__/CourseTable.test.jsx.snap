// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseTable should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseTable
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello Publisher!",
    },
    "ref": null,
    "rendered": "Hello Publisher!",
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello Publisher!",
      },
      "ref": null,
      "rendered": "Hello Publisher!",
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
